test/clj/matcher_combinators/matchers_test.clj

```diff
@@ -395,3 +395,29 @@
     (is (match? (m/in-any-order
                  [odd? pos? (m/mismatch odd?)])
                 [1 2 3]))))
+
+(deftest via-matcher
+  (testing "normal usage"
+    (is (match? {:payloads [(m/via read-string {:foo :bar})]}
+                {:payloads ["{:foo :bar}"]})))
+
+  (testing "via + match-with allows pre-processing `actual` before applying matching"
+    (is (match? (m/match-with
+                 [vector? (fn [expected] (m/via reverse expected))]
+                 {:payloads [1 2 3]})
+                {:payloads [3 2 1]})))
```

Comment:
this test is a contrived example but [here is a more realistic use case ](https://github.com/clj-kondo/clj-kondo/issues/1869#issuecomment-1308556698)

Reply:
oh, doing `(m/match-with [vector? (fn [expected] (m/via sort expected))] ...)` is actually a pretty useful example (much more than `reverse` here)